pkg/operators/filter/filter.go

```diff
@@ -100,21 +113,73 @@ func (f *filterOperator) InstantiateDataOperator(gadgetCtx operators.GadgetConte
 		}
 	}
 
+	for _, ds := range gadgetCtx.GetDataSources() {
```

Comment:
Maybe store `gadgetCtx.GetDataSources()` in a local variable and use it also for `keyForDataSource`?